openmc/model/surface_composite.py

```diff
@@ -773,15 +763,140 @@ def _make_ccw(self, points):
         -------
         ordered_points : the input points ordered counter-clockwise
         """
+        points = np.asarray(points, dtype=float)
+        check_iterable_type('points', points, float, min_depth=2, max_depth=2)
+        check_length('points', points[0, :], 2, 2)
+
+        # If the last point is the same as the first, remove it and make sure
+        # there are still at least 3 points for a valid polygon.
+        if np.allclose(points[0, :], points[-1, :]):
+            points = points[:-1, :]
+        check_length('points', points, 3)
+
+        if len(points) != len(np.unique(points, axis=0)):
+            raise ValueError('Duplicate points were detected in the Polygon input')
+
+        # Order the points counter-clockwise (necessary for offset method)
         # Calculates twice the signed area of the polygon using the "Shoelace
         # Formula" https://en.wikipedia.org/wiki/Shoelace_formula
+        # If signed area is positive the curve is oriented counter-clockwise.
+        # If the signed area is negative the curve is oriented clockwise.
         xpts, ypts = points.T
+        if np.sum(ypts*(np.roll(xpts, 1) - np.roll(xpts, -1))) < 0:
+            points = points[::-1, :]
+
+
+        # Check if polygon is self-intersecting by comparing edges pairwise
```

Comment:
If there is a source for the below method? If so, it might be worth adding a link such that potential debugging in the future is easier.

Reply:
It's loosely based on the link I've added as a comment